src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlAction.cs

```diff
@@ -25,42 +25,85 @@
 //
 //------------------------------------------------------------------------------
 
-using Microsoft.IdentityModel.Logging;
+using System;
+using static Microsoft.IdentityModel.Logging.LogHelper;
 
 namespace Microsoft.IdentityModel.Tokens.Saml
 {
+    /// <summary>
+    /// Represents the Action element specified in [Saml, 2.4.5.1].
+    /// </summary>
     public class SamlAction
     {
-        string _ns;
-        string _action;
+        private Uri _namespace;
+        private string _value;
 
+        /// <summary>
+        /// Constructs an instance of SamlAction class.
+        /// </summary>
         internal SamlAction() { }
 
+        /// <summary>
+        /// Constructs an instance of Saml2Action class.
+        /// </summary>
+        /// <param name="action">Action value represented by this class.</param>
         public SamlAction(string action)
             : this(action, null)
         {
         }
 
-        public SamlAction(string action, string ns)
+        /// <summary>
+        /// Constructs an instance of SamlAction class.
+        /// </summary>
+        /// <param name="action">Action value represented by this class.</param>
+        /// <param name="actionNamespace">Namespace in which the action is interpreted.</param>
+        public SamlAction(string action, Uri actionNamespace)
         {
             if (string.IsNullOrEmpty(action))
-                throw LogHelper.LogArgumentNullException(nameof(action));
+                throw LogArgumentNullException(nameof(action));
 
-            if (string.IsNullOrEmpty(ns))
-                throw LogHelper.LogArgumentNullException(nameof(ns));
-
-            _action = action;
-            _ns = ns;
+            _value = action;
+            _namespace = actionNamespace ?? new Uri(SamlConstants.DefaultActionNamespace);
         }
 
-        public string Action
+        /// <summary>
+        /// Gets or sets a URI reference representing the namespace in which the name of the
+        /// specified action is to be interpreted. [Saml, 2.4.5.1]
+        /// </summary>
+        public Uri Namespace
         {
-            get { return _action; }
+            get
+            {
+                return _namespace;
+            }
+
+            set
+            {
+                if (!value.IsAbsoluteUri)
```

Comment:
_namespce is never set to the value. The logic used to be if value is null, set to default. If not null, make sure it is an absolute Uri.